dvc/dagascii.py

```diff
@@ -12,6 +15,45 @@
 from grandalf.routing import EdgeViewer
 from grandalf.routing import route_with_lines
 
+from dvc.env import DVC_PAGER
+
+
+logger = logging.getLogger(__name__)
+
+
+DEFAULT_PAGER = "less"
+DEFAULT_PAGER_FORMATTED = "{} --chop-long-lines --clear-screen".format(
+    DEFAULT_PAGER
+)
+
+
+def find_pager():
+    if not sys.stdout.isatty():
+        return pydoc.plainpager
+
+    env_pager = os.getenv(DVC_PAGER)
+    if env_pager:
+
+        def user_pager(text):
+            return pydoc.tempfilepager(pydoc.plain(text), env_pager)
+
+        return user_pager
+    else:
+        if os.system("({}) 2>{}".format(DEFAULT_PAGER, os.devnull)) == 0:
+
+            def less_pager(text):
+                return pydoc.tempfilepager(
+                    pydoc.plain(text), DEFAULT_PAGER_FORMATTED
+                )
+
+            return less_pager
+
```

Comment:
Let's do something like
```
    def make_pager(cmd):
        def pager(text):
            return pydoc.tempfilepager(pydoc.plain(text), cmd)
        return pager

    env_pager = os.getenv(DVC_PAGER)
    if env_pager:
        return make_pager(env_pager)
    elif os.system("({}) 2>{}".format(DEFAULT_PAGER, os.devnull)) == 0:
        return make_pager(DEFAULT_PAGER_FORMATTED)

    logger.warning()
    return pydoc.plainpager
```

This way there is less "if complexity" and less lines of code 🙂

Reply:
good tip, thanks!